src/components/CurrencyInputPanel/FiatValue.tsx

```diff
@@ -35,11 +36,19 @@ export function FiatValue({
     return <FiatLoadingBubble />
   }
 
+  let formattedValue = formatNumber(fiatValue.data, NumberType.FiatTokenPrice)
+  if (formattedValue.length > 18) {
+    //Removes the dollar sign at the start, commas, and decimal places
+    formattedValue = formattedValue.slice(1).replaceAll(',', '').slice(0, -4)
+    formattedValue = formatTransactionAmount(Number(formattedValue))
+    formattedValue = '$' + formattedValue + 'T'
```

Comment:
Bringing in @JackShort as this logic may not work for internationalization of currencies